pithos/plugins/notification_icon.py

```diff
@@ -112,6 +79,18 @@ def check_for_plasma():
                 None
             )
 
+        def on_StatusNotifierWatcher_finish(source, result, data):
+            try:
+                StatusNotifierWatcher = Gio.DBusProxy.new_finish(result)
+            except GLib.Error as e:
+                logging.debug(e)
+                check_for_x()
+            else:
+                if StatusNotifierWatcher.get_name_owner():
+                    init_appindicator()
+                else:
+                    check_for_plasma()
```

Comment:
I'm not sure why it all couldn't be one interface it seems really stupid that there are 3 interfaces since all are required to get the desired end result StatusNotifierHost seems especially useless...